static/app/components/sidebar/index.tsx

```diff
@@ -118,7 +119,9 @@ function Sidebar({organization}: Props) {
 
   const collapsed = !!preferences.collapsed;
   const horizontal = useMedia(`(max-width: ${theme.breakpoints.medium})`);
-  const hasSuperuserSession = isActiveSuperuser();
+
+  // Avoid showing superuser UI on self-hosted instances
+  const hasSuperuserSession = isActiveSuperuser() && !ConfigStore.get('isSelfHosted');
```

Comment:
I was surprised to see `isActiveSuperuser` in sentry repo and not getsentry.

Reply:
self-hosted also uses superuser for instance level administration views and endpoints.

Reply:
oh

@leedongwei should the warning/alert be a hook? We could have getsentry implement the "you'll get fired" part, and for self-hosted and other contexts the alert still be red across the top and say 'You are in superuser mode.'

Reply:
@ryan953 Yea it's been moved into a hook.